src/plugins/dbus/README.md

```diff
@@ -187,3 +187,33 @@ of the whole system knows that a chain of reactions will terminate.
 When doing such event-driven programming, care is needed to avoid
 infinite loops.  Elektra guarantees consistency of the key database even
 in such cases.
+
+# Transport Plugin
+
+Mount this plugin globally with default settings to use it as *sending*
+transport plugin for Elektra's notification feature:
+
+> kdb global-mount dbus
+
+# Notification Format
+
+This plugin uses D-Bus signal messages as notifications.
+Notifications we path `/org/libelektra/configuration` and the D-Bus interface
+`org.libelektra`.
+The following signal names are used:
+
+- KeyAdded: a key has been added
+- KeyChanged: a key has been changed
+
+The first argument contains the name of the changed key.
+The system bus is used if the affected keys is below the `system` namespace.
+If the key is below the `user` namespace the session bus is used.
+
+Example output from `dbus-monitor`:
+
+```
+signal time=1519856379.538736 sender=:1.1 -> destination=(null destination) serial=18 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyAdded
+   string "system/tests/foo/bar"
+signal time=1519856379.538753 sender=:1.1 -> destination=(null destination) serial=19 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyChanged
```

Comment:
Is this information used within your internal notification?

Otherwise we already said we do not need a notification for every key individually. See #1134

Reply:
Ok that wasn't very clear when reading the conversation. Currently when a notification arrives `kdbGet()` is executed with this key. I plan to add a check if the application has registered a key below the changed key. The type of change is actually not used.

So I see no problem using/supporting the "Commit" announcement of the dbus plugin.

Reply:
> keyNeedSync

Unfortunately this approach is buggy, see #404

> Currently when a notification arrives kdbGet() is executed with this key.

So it might lead to a massive amount of kdbGet()?

> So I see no problem using/supporting the "Commit" announcement of the dbus plugin.

Commit messages are necessary, even when sending only a single message per backend the notification system somehow needs to know when to trigger a reload (when all backends are executed). A proper design for this is badly needed.

Reply:
Ok, so the change detection path is a minefield :)

> So it might lead to a massive amount of kdbGet()?

In the current state yes but I will add the necessary changes to this PR. I had some doubts about overwritten keys and cascading, but kdbGet() should handle these fine.

I thought for global plugins the `postcommit` position ensures that the backends are done and the data is final. A "Commit" message on kdbSet() with the parentKey would notify other applications about changes below this key.